### ISSUE TITLE
'The operation has timed out' during AppService build stage

### ISSUE THREAD

**Issue by psfinaki** [SAFE-Stack/SAFE-template#120]
Does everyone manage to upload the app service content to Azure with default `WebClient` in 100% cases?
I have quite fast internet and still nearly always, with the default build script, I get this exception - very sad to get it at the last stage of the script.

I know a work around. Default .Net `WebClient` has some 100 seconds of timeout and [this](https://stackoverflow.com/a/6994391/3232646) is a way to overwrite it. Ported to F#:
```fsharp
type WebClient'() = 
    inherit Net.WebClient()
    override this.GetWebRequest uri = 
        let request = base.GetWebRequest uri
        request.Timeout <- 30 * 60 * 1000
        request

Target.create "AppService" (fun _ ->
    let zipFile = "deploy.zip"
    IO.File.Delete zipFile
    Zip.zip deployDir zipFile !!(deployDir + @"\**\**")

    let appName = deploymentOutputs.Value.WebAppName.value
    let appPassword = deploymentOutputs.Value.WebAppPassword.value

    let destinationUri = sprintf "https://%s.scm.azurewebsites.net/api/zipdeploy" appName
    let client = new WebClient'(Credentials = Net.NetworkCredential("$" + appName, appPassword))
    Trace.tracefn "Uploading %s to %s" zipFile destinationUri
    client.UploadData(destinationUri, IO.File.ReadAllBytes zipFile) |> ignore)
```
Half an hour is definitely enough, this works :)

I can create a pull request and probably write a little in the docs on that. 
Or does anybody know a better way to go?

**Comment by theimowski** [comment]
I never got a timeout on AppService yet.
Please send a PR - this workaround is fine
/cc @isaacabraham

**Comment by isaacabraham** [comment]
I've actually seen this before, although it's never happened to me personally. And yes, creating a web client without timeout was the same fix we applied. I believe that it is somehow environmental but don't know more than that.

**Comment by isaacabraham** [comment]
Obviously will be happy to accept a PR on this.

**Comment by psfinaki** [comment]
Alright, so, you have pull request :)

**Comment by theimowski** [comment]
Resolved by #122
